hpy/devel/include/common/hpydef.h

```diff
@@ -70,11 +70,16 @@ typedef struct {
     void *closure;
 } HPyGetSet;
 
+typedef struct {
+    const char *docstring;  // the null-terminated UTF-8 docstring
+} HPyDocString;
```

Comment:
Why do you create a separate definition kind?
Since it's just a slot in CPython, we could also just store it in `HPySlot`. This, of course, has the penalty of two more words (`HPySlot.cpy_trampoline` and the slot kind).

Reply:
well, in CPython everything is a slot :). In HPy we are trying to be a more type safe so if we want to introduce `HPy_tp_doc`, I think this is the way to go.
(I still think I prefer `.doc`, though)

Reply:
HPy works a bit differently. The macros generate `HPyDef` objects, which are then examined when the type is created. I could stuff the docstring into the `void * impl` filed of HPySlot, but then all the logic that looks at the slots would have to special case the docstring because it's not a function.